pkg/tasks/prometheus_user_workload.go

```diff
@@ -347,7 +364,33 @@ func (t *PrometheusUserWorkloadTask) destroy(ctx context.Context) error {
 		}
 	}
 
-	p, err := t.factory.PrometheusUserWorkload(s)
+	trustedCA, err := t.factory.PrometheusUserWorkloadTrustedCABundle()
+	if err != nil {
+		return errors.Wrap(err, "initializing UserWorkload CA bundle ConfigMap failed")
+	}
+
+	cbs := &caBundleSyncer{
+		client:  t.client,
+		factory: t.factory,
+		prefix:  "prometheus-user-workload",
+	}
+
+	hashedTrustedCA, err := cbs.syncTrustedCABundle(ctx, trustedCA)
```

Comment:
instead of using caBundleSyncer, it's better to use `DeleteHashedConfigMap` like here

https://github.com/openshift/cluster-monitoring-operator/blob/88d97c857d398cb18a697f2d0dffa6acf64825d4/pkg/tasks/thanos_ruler_user_workload.go#L384-L397